keps/77-dynamically-sized-jobs/README.md

```diff
@@ -0,0 +1,286 @@
+2. Call *updateWorkloadToMatchJob()*, this will construct and update the workload and in turn update the PodSet Count field.
+3. Inside the *Update()* method from the *workload_controller* update the workload in cache and in queue. By updating the workload in cache this will update the cluster queue resource usage and by updating the workload in queue this will trigger the scheduler so that it re-assigns the flavors to the already assumed workload and in this way PodSetAssignments will be updated by applying admission based on the new assignments. 
+4. Inside the schedule logic in the scheduler, since the workload is already assumed in the cache we need to specify if the feature is enabled so that we can apply admission to the workload and update its PodSetAssignments. 
+
+#### Job controller
+
+Rework *equivalentToWorkload()* and *reconcile()* to account for potential differences between the job’s number of replicas and the running workload’s PodSet.Count. 
+
+Given these changes, check if the delta is positive or negative indicating a scaleup/scaledown. If it’s a scaledown the reconciler should trigger an update on the workload to match the new job’s spec and no quota needs to be checked or accounted for, the cluster queue should update the workload resource usage. 
+
+### Phase 2 - Aggregating Workload Slices
+
+In Phase 2, aggregating Workload Slices into the parent workload will be implemented.
+
+### Phase 3 - Scale up with Workload Slices and Scheduling Gates
+
+In Phase 3, scale up will be implemented by introducing Workload Slices and adding Pod scheduling gates as part of Kueue’s mutating admission for RayCluster. 
+
+When the RayCluster scales, the RayCluster webhook would be modified to intercept and "gate" all pods. Every time there’s a resize, you create a dependable (child) workload slice and once it's admitted, you increase the count in the original workload, delete the old workload and remove the schedulingGates. 
+- Pros: You are able to hold the pods added by Kuberay
+- Cons: The fact of having schedulingGates, means we need an API call for every pod, because all pods that are created by the RayCluster are going to have schedulingGates. We need to remove those gates and for every pod you need to make API calls.
+
+#### Scheduler
+Since every scale up will have its own individual workload they should be proposed to the current scheduling cycle and continue the normal admission process. We should lock flavor assignments to Workload Slices we need to ensure that the Workload Slice is assigned the same resource flavor as the parent workload. 
+
+The *nominate()* returns the workloads with their requirements (resource flavors, borrowing) if they were admitted by the clusterQueues in the snapshot, so we need to return the original workload that was already admitted with the resize information inside *TotalRequests* so that *PodSetAssignments* is also updated.
```

Comment:
```suggestion
The `nominate()` returns the workloads with their requirements (resource flavors, borrowing) if they were admitted by the clusterQueues in the snapshot, so we need to return the original workload that was already admitted with the resize information inside *TotalRequests* so that *PodSetAssignments* is locked to the parent Workload assignments.
```